nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/security/ACE.java

```diff
@@ -69,11 +71,24 @@ public static ACE fromId(String aceId) {
             return null;
         }
 
-        String[] parts = aceId.split(":");
-        if (parts.length < 3) {
-            throw new IllegalArgumentException(String.format("Invalid ACE id: %s", aceId));
+	// An ACE is composed of tokens separated with ":" caracter
+	// First 3 tokens are mandatory; following 3 tokens are optional
+        ArrayList<String> tokenList = new ArrayList<>();
+	Pattern p = Pattern.compile("^(.+):([^:]+):([^:]+):([^:]*):([^:]*):([^:]*)$");
+        Matcher m = p.matcher(aceId);
+        boolean b = m.matches();
+        if(!b) {
```

Comment:
Add space between `if` and `(`. No need for a separate `b` variable, just do `if (!m.matches())`